File: app/src/main/java/com/san/moviedbclientmvc/ui/movieslist/view/controller/MoviesListViewContract.java
package com.san.moviedbclientmvc.ui.movieslist.view.controller;

import com.san.moviedbclientmvc.networking.model.Movie;
import com.san.moviedbclientmvc.common.views.ObservableViewMvc;

import java.util.List;

public interface MoviesListViewContract extends ObservableViewMvc<MoviesListViewContract.Listener> {

    public interface Listener {
        void onMovieItemClicked(Movie movie);
    }

    void bindMovies(List<Movie> movies);

    void showProgressIndication();

    void hideProgressIndication();

}

File: app/src/main/java/com/san/moviedbclientmvc/ui/movieslist/view/controller/MoviesListViewContract.kt
package com.san.moviedbclientmvc.ui.movieslist.view.controller

import android.os.Parcelable
import com.san.moviedbclientmvc.common.views.ObservableViewMvc
import com.san.moviedbclientmvc.networking.model.Movie

interface MoviesListViewContract: ObservableViewMvc<MoviesListViewContract.Listener> {

    interface Listener {
        fun onMovieItemClicked(movie: Movie)
        fun saveOnBackPressed(parcelable: Parcelable): Parcelable
    }

    fun bindMovies(movies: List<Movie>)
    fun showProgressIndication()
    fun hideProgressIndication()
}